src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -658,14 +644,12 @@ public void ProcessCommand(IExecutionContext context, string inputLine, ActionCo
                 }
             }
 
-            foreach (var property in command.Properties)
-            {
-                if (!string.Equals(property.Key, Constants.Runner.InternalTelemetryIssueDataKey, StringComparison.OrdinalIgnoreCase))
-                {
-                    issue.Data[property.Key] = property.Value;
-                }
-            }
+            string keyToExclude = Constants.Runner.InternalTelemetryIssueDataKey;
+            var filteredDictionaryEntries = command.Properties
+                                                   .Where(kvp => !string.Equals(kvp.Key, keyToExclude, StringComparison.OrdinalIgnoreCase));
```

Comment:
I probably need to close over this `IEnumerable` since there's no guarantee that `command` won't get modified by the time the `IssueMetadata` is consumed.

Reply:
fixed